src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -14,107 +15,67 @@ class er;
 /// \endcond
 
 /// Numerical integration algorithms
-namespace integrate {
-
-/// The integration result is returned in this format.
-struct Result {
-  double value;
-  double error;
-};
+namespace integration {
 
 namespace detail {
-template <typename... Args>
-class GslQuadAdaptiveImpl;
-
-template <typename Args, typename ArgsIndices =
-                             std::make_index_sequence<tmpl::size<Args>::value>>
-struct ExpandTupleImpl;
-
-template <typename... Args, size_t... ArgsIndices>
-struct ExpandTupleImpl<tmpl::list<Args...>,
-                       std::index_sequence<ArgsIndices...>> {
-  template <typename FunctionType>
-  static double apply(FunctionType&& function, const double x,
-                      const std::tuple<Args...>& args) {
-    return (*function)(x, std::get<ArgsIndices>(args)...);
-  }
-};
-
-template <typename... Args>
-double integrand(double x, void* params) {
-  auto args = reinterpret_cast<GslQuadAdaptiveImpl<Args...>*>(params);
-  return ExpandTupleImpl<tmpl::list<Args...>>::apply(args->function_, x,
-                                                     args->args_);
-}
 
 // The GSL functions require the integrand to have this particular function
 // signature. In particular, any extra parameters to the functions must be
-// passed as a void* and re-interpreted appropriately. We interpret the
-// pointer as an instance of GslQuadAdaptiveImpl and use it to retrieve the
-// integrand function and its extra parameters.
-// The class builds a numerical integrator with the maximum number of
+// passed as a void* and re-interpreted appropriately.
+// GslQuadAdaptiveImpl builds a numerical integrator with the maximum number of
 // subintervals passed into the constructor.
```

Comment:
delete the last two lines, they have nothing to do with this function

Reply:
Done.